### ISSUE TITLE
MS Edge: clicking a link in a sidebar mod crashes MS Edge

### ISSUE THREAD

**Issue by twschiller** [pixiebrix/pixiebrix-extension#8206]
Steps to Reproduce
---
- Click the "Open Google Sheet" link in the Announcements/Notices Mod Sidebar

Actual Behavior
---
- MS Edge crashes
- PixieBrix no longer installed (might only impact load unpacked?)

Discussion
---
- In previous PRs, we had fixed native link behavior: https://github.com/pixiebrix/pixiebrix-extension/pull/7889. This might be a regression? An issue stemming from use of Shadow DOM if the handler can't intercept it?

Playwright Test Scenarios (can be put in a single test mod)
---
- Click the gear to open the Extension Console
- Use a test mod with the following: `@pixies/test/sidebar-links`
  - Open in tab brick
  - Link in markdown rendered in text element
  - Link in embedded HTML brick
  - Link in embedded form: form description and field description support markdown. Likely only need to test 1

**Comment by fregante** [comment]
The panel's content is double-wrapped in the Shadow DOM, so the final element does not reach it.

I think this is an argument to keep the shadow roots `open`, at least in when used in the sidebar. If you merge this, it will probably fix this issue by changing the default `mode` for `IsolatedComponent` to `open`:

- https://github.com/pixiebrix/pixiebrix-extension/pull/8151

**Comment by twschiller** [comment]
PR merged. Remaining work is to add playwright test for the scenarios

**Comment by twschiller** [comment]
@fregante how was the PR tested? On https://github.com/pixiebrix/pixiebrix-extension/actions/runs/8638318978 the gear icon isn't doing anything for me. It doesn't change style on hover and isn't responding to clicks

**Comment by fregante** [comment]
Commented in your PR

**Comment by twschiller** [comment]
I've submitted a minimal reproduction to the MS Edge team: https://github.com/microsoft/MicrosoftEdge-Extensions/issues/145

**Comment by twschiller** [comment]
Closing as won't fix -- we're waiting on MS Edge to address because MS Edge's content script injection also is different

**Comment by twschiller** [comment]
@grahamlangford a fix is in the MS Edge canary build: https://github.com/microsoft/MicrosoftEdge-Extensions/issues/145#issuecomment-2270424416
